rust/abacus-base/src/settings/chains.rs

```diff
@@ -291,3 +295,122 @@ impl ChainSetup<InboxAddresses> {
         cfg
     }
 }
+
```

Comment:
I added these to add some compile-time checks on ensuring we always have a coingecko API ID for each mainnet chain. I'm curious how you feel about this vs the existing macro here: https://github.com/abacus-network/abacus-monorepo/blob/main/rust/abacus-core/src/chain.rs#L78

If you like this approach I'll move over the macro linked above so we only have one source of truth for abacus chains / domains, but otherwise I'm happy to get rid of this. Another idea is instead of having separate enums for mainnet / testnet / local test chains, is to just have one big enum with them all, and then have a runtime check for getting coingecko API IDs.

Just generally would love feedback here because I could see this not being very ergonomic